### PR TITLE
[Breaking] Modify return type of created_at/finished_at to datetime

### PR DESCRIPTION
**Pull request by savingoyal**
Metaflow Client currently returns a string for `created_at` and `finished_at` properties. This PR changes the response type to a more appropriate Datetime object.

### CODE REVIEW

File: metaflow/client/core.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from __future__ import print_function
+from datetime import datetime
 import os
 import time
```
Comment (romain-intel) [inline metaflow/client/core.py:4]: Nit: can remove since no longer used.

**Review by romain-intel — APPROVED** [review]
Minor nit but otherwise LGTM.

File: metaflow/client/core.py (hunk 1 above)
Comment (savingoyal) [reply metaflow/client/core.py]: Done

**Comment by savingoyal** [comment]
I will merge this PR right when we are merging #580

**Pull request merged**